inc/cobs.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdint>
+#include <cstddef>
+
+/*
+Consistent Overhead Byte Stuffing (COBS) removes a specific value from a
+list of values. The removed value can then, for example, be used as a
+frame marker.
+
+Refer to: https://en.wikipedia.org/wiki/Consistent_Overhead_Byte_Stuffing
+
+COBS considerations:
+
+1. Configurable marker byte value. The current implementation assumes
+0x00.
+
+2. If the encoding results in { ..., 0, 1 }, encode it as { ..., 0 }.
+The current implementation applies this reduced encoding.
+
+3. If the encoding results in { n, ..., x, 0 } where n is an offset and
+x > n, encode it as { x, ..., 0 }. The standard implementation would run
+into an decoding error because x > n is bigger than the remaining number
+of bytes. The current implementation does not apply this reduced
+encoding.
+
+4. Append the marker byte to the encoded result. The current
+implementation does append the marker byte.
+
+Here are some interesting cases. x is a non-zero byte. 0 is
+a zero byte. 123x denotes 123 consecutive x's.
+
+decoded, length -> encoded, length, note
+       , 0      -> 0      , 1     , short encoding
+       , 0      -> 1|0    , 2     , standard encoding
```

Comment:
Because the presence of a packet is also information, even if the contents are empty. If you don' t want to send empty packets then don' t call cobs::encode.